gemrb/core/Map.cpp

```diff
@@ -2475,141 +2441,72 @@ void Map::AdjustPosition(Point &goal, unsigned int radiusx, unsigned int radiusy
-		long ty = (long) y - goal.y;
-		unsigned int distance = (unsigned int) std::sqrt( ( double ) ( tx* tx + ty* ty ) );
-		if (dist<distance) {
-			best.x=(ieWord) x;
-			best.y=(ieWord) y;
-			dist=distance;
-		}
-
-		unsigned int Cost = MapSet[y * Width + x] + NormalCost;
-		if (Cost > PathLen) {
-			break;
-		}
-		SetupNode( x - 1, y - 1, size, Cost );
-		SetupNode( x + 1, y - 1, size, Cost );
-		SetupNode( x + 1, y + 1, size, Cost );
-		SetupNode( x - 1, y + 1, size, Cost );
-
-		Cost += AdditionalCost;
-		SetupNode( x, y - 1, size, Cost );
-		SetupNode( x + 1, y, size, Cost );
-		SetupNode( x, y + 1, size, Cost );
-		SetupNode( x - 1, y, size, Cost );
-	}
-
-	//find path backwards from best to start
-	PathNode* StartNode = new PathNode;
-	PathNode* Return = StartNode;
-	StartNode->Next = NULL;
-	StartNode->x = best.x;
-	StartNode->y = best.y;
-	if (flags) {
-		StartNode->orient = GetOrient( start, best );
-	} else {
-		StartNode->orient = GetOrient( best, start );
-	}
-	Point p = best;
-	unsigned int pos2 = start.y * Width + start.x;
-	while (( pos = p.y * Width + p.x ) != pos2) {
-		Return = new PathNode;
-		StartNode->Parent = Return;
-		Return->Next = StartNode;
-		StartNode = Return;
-		unsigned int level = MapSet[pos];
-		unsigned int diff = 0;
-		Point n;
-		Leveldown( p.x, p.y + 1, level, n, diff );
-		Leveldown( p.x + 1, p.y, level, n, diff );
-		Leveldown( p.x - 1, p.y, level, n, diff );
-		Leveldown( p.x, p.y - 1, level, n, diff );
-		Leveldown( p.x - 1, p.y + 1, level, n, diff );
-		Leveldown( p.x + 1, p.y + 1, level, n, diff );
-		Leveldown( p.x + 1, p.y - 1, level, n, diff );
-		Leveldown( p.x - 1, p.y - 1, level, n, diff );
-		Return->x = n.x;
-		Return->y = n.y;
-
-		if (flags) {
-			Return->orient = GetOrient( p, n );
-		} else {
-			Return->orient = GetOrient( n, p );
-		}
-		p = n;
-		if (!diff) {
-			break;
+//the 5th parameter is controlling the orientation of the actor
+//0 - back away, 1 - face direction
+PathNode* Map::RunAway(const Point &s, const Point &d, unsigned int size, unsigned int PathLen, int noBackAway)
+{
+	SearchmapPoint smptFarthest = FindFarthest(d, size, PathLen);
+	NavmapPoint nmptFarthest = NavmapPoint(smptFarthest.x * 16, smptFarthest.y * 12);
+
+	return FindPath(s, nmptFarthest, size, 0, true, !noBackAway);
+}
+
+SearchmapPoint Map::FindFarthest(const NavmapPoint &d, unsigned int size, unsigned int pathLength, int validFlags) const
+{
+	float *dist = new float[Width * Height]();
+	static const size_t DEGREES_OF_FREEDOM = 8;
+	static const char dx[DEGREES_OF_FREEDOM] = {1, 0, -1, 0, 1, 1, -1, -1};
+	static const char dy[DEGREES_OF_FREEDOM] = {0, 1, 0, -1, 1, -1, 1, -1};
+	SearchmapPoint smptFleeFrom = SearchmapPoint(d.x / 16, d.y / 12);
+	std::priority_queue<PQNode> open;
+	static const float diagWeight = sqrt(2);
+	dist[smptFleeFrom.y * Width + smptFleeFrom.x] = 0;
+	open.push(PQNode(smptFleeFrom, 0));
+	SearchmapPoint smptFarthest;
+	SearchmapPoint smptCurrent = smptFleeFrom;
```

Comment:
Codacy found an issue: [Variable 'smptCurrent' is assigned a value that is never used.](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=3427588)